lib/improver/tests/utilities/test_pad_spatial.py

```diff
@@ -0,0 +1,384 @@
+        self.cube.data[0, 2, 2] = 0
+
+    def test_yx_order(self):
+        """Test that a cube is created with the expected order for the y and x
+        coordinates within the output cube, if the input cube has dimensions
+        of projection_y_coordinate and projection_x_coordinate."""
+        for sliced_cube in self.cube.slices(
+                ["projection_y_coordinate", "projection_x_coordinate"]):
+            break
+        data = sliced_cube.data
+        coord_x = sliced_cube.coord("projection_x_coordinate")
+        coord_y = sliced_cube.coord("projection_y_coordinate")
+
+        new_cube = create_cube_with_new_data(
+            sliced_cube, data, coord_x, coord_y)
+        self.assertIsInstance(new_cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(new_cube.data, data)
+        self.assertEqual(new_cube.coord_dims("projection_y_coordinate")[0], 0)
+        self.assertEqual(new_cube.coord_dims("projection_x_coordinate")[0], 1)
+
+    def test_xy_order(self):
+        """Test that a cube is created with the expected order for the y and x
+        coordinates within the output cube, if the input cube has dimensions
+        of projection_x_coordinate and projection_y_coordinate."""
+        for sliced_cube in self.cube.slices(
+                ["projection_x_coordinate", "projection_y_coordinate"]):
+            break
+        data = sliced_cube.data
+        coord_x = sliced_cube.coord("projection_x_coordinate")
+        coord_y = sliced_cube.coord("projection_y_coordinate")
+
+        new_cube = create_cube_with_new_data(
+            sliced_cube, data, coord_x, coord_y)
+        self.assertIsInstance(new_cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(new_cube.data, data)
+        self.assertEqual(new_cube.coord_dims("projection_y_coordinate")[0], 1)
+        self.assertEqual(new_cube.coord_dims("projection_x_coordinate")[0], 0)
+
+    def test_realization(self):
+        """Test that a cube is created with the expected order for coordinates
+        within the output cube, if the input cube has dimensions of
+        realization, projection_y_coordinate and projection_x_coordinate."""
+        for sliced_cube in self.cube.slices(["realization",
+                                             "projection_y_coordinate",
+                                             "projection_x_coordinate"]):
+            break
+        data = sliced_cube.data
+        coord_x = sliced_cube.coord("projection_x_coordinate")
+        coord_y = sliced_cube.coord("projection_y_coordinate")
+
+        new_cube = create_cube_with_new_data(
+            sliced_cube, data, coord_x, coord_y)
+        self.assertIsInstance(new_cube, iris.cube.Cube)
+        self.assertArrayAlmostEqual(new_cube.data, data)
+        self.assertEqual(new_cube.coord_dims("realization")[0], 0)
+        self.assertEqual(new_cube.coord_dims("projection_y_coordinate")[0], 1)
+        self.assertEqual(new_cube.coord_dims("projection_x_coordinate")[0], 2)
+
```

Comment:
There is a forecast period coordinate on the input cube for every test now.  I can check that it's unchanged, if you like?  I can't see why it would be.

Is the need for a test to show that no other coordinates or attributes (apart from the spatial coordinates and data dimensions) are changed by this function?

Reply:
Removing this unit test is fine.